pkg/covermerger/covermerger.go

```diff
@@ -50,6 +54,95 @@ type FileCoverageMerger interface {
 	Result() *MergeResult
 }
 
+// MergeCSVWriteJSONL mergers input CSV and generates JSONL records.
+// The amount of lines generated is [count(managers)+1] * [count(kernel_files)].
+// Returns (totalInstrumentedLines, totalCoveredLines, error).
+func MergeCSVWriteJSONL(config *Config, descr *coveragedb.HistoryRecord, csvReader io.Reader, w io.Writer,
+) (int, int, error) {
+	eg := errgroup.Group{}
+	mergeResults := make(chan *FileMergeResult)
+	eg.Go(func() error {
+		defer close(mergeResults)
+		if err := MergeCSVData(config, csvReader, mergeResults); err != nil {
+			return fmt.Errorf("covermerger.MergeCSVData: %w", err)
+		}
+		return nil
+	})
+	var totalInstrumentedLines, totalCoveredLines int
+	eg.Go(func() error {
+		var encoder *json.Encoder
+		if w != nil {
+			gzw := gzip.NewWriter(w)
+			defer gzw.Close()
+			encoder = json.NewEncoder(gzw)
+		}
+		if encoder != nil {
+			if err := encoder.Encode(descr); err != nil {
+				return fmt.Errorf("encoder.Encode(MergedCoverageDescription): %w", err)
+			}
+		}
+		for fileMergeResult := range mergeResults {
+			dashCoverageRecords := mergedCoverageRecords(fileMergeResult)
+			if encoder != nil {
+				for _, record := range dashCoverageRecords {
+					if err := encoder.Encode(record); err != nil {
+						return fmt.Errorf("encoder.Encode(MergedCoverageRecord): %w", err)
+					}
+				}
+			}
+			for _, hitCount := range fileMergeResult.HitCounts {
+				totalInstrumentedLines++
+				if hitCount > 0 {
+					totalCoveredLines++
+				}
+			}
+		}
+		return nil
+	})
+	if err := eg.Wait(); err != nil {
+		return 0, 0, fmt.Errorf("eg.Wait: %w", err)
+	}
+	return totalInstrumentedLines, totalCoveredLines, nil
+}
+
+const allManagers = "*"
+
+func mergedCoverageRecords(fmr *FileMergeResult,
```

Comment:
I think we shouldn't be breaking the line here.

Reply:
Thanks. Done.
I don't see how can we catch it with golangci-lint... Let me spend 10 more minutes for it...

Reply:
One more custom linter may help here...